lib/services/saml.go

```diff
@@ -266,7 +270,7 @@ func GetSAMLServiceProvider(sc types.SAMLConnector, clock clockwork.Clock) (*sam
 }
 
 // UnmarshalSAMLConnector unmarshals the SAMLConnector resource from JSON.
-func UnmarshalSAMLConnector(bytes []byte, opts ...MarshalOption) (types.SAMLConnector, error) {
+func UnmarshalSAMLConnector(ctx context.Context, bytes []byte, opts ...MarshalOption) (types.SAMLConnector, error) {
```

Comment:
Interesting how the context suddenly exposes a bunch of methods that, naively, I wouldn't expect to depend on networking.

Reply:
Yeah, as what started as a small little fix starts to sprawl, I'm wondering if we're even doing this right. Every time it validates, it fetches the entity descriptor if there is a URL there, and updates the entity descriptor field with the result. But I'm not sure that gets stored every time (likely not), so if the results of the GET change, things might get a little weird with hard to reproduce results. But this code has been there for 5 years so I guess it hasn't caused any noticable problems.